python.d/monit.chart.py

```diff
@@ -0,0 +1,146 @@
+retries = 60
+
+# see enum State_Type from monit.h (https://bitbucket.org/tildeslash/monit/src/master/src/monit.h)
+MONIT_SERVICE_NAMES = [ 'Filesystem', 'Directory', 'File', 'Process', 'Host', 'System', 'Fifo', 'Program', 'Net' ]
+DEFAULT_SERVICES_IDS = [ 0, 1, 2, 3, 4, 6, 7, 8 ]
+
+# charts order (can be overridden if you want less charts, or different order)
+ORDER = [ 'filesystem', 'directory', 'file', 'process', 'process_uptime', 'process_threads', 'process_children', 'host', 'host_latency', 'system', 'fifo', 'program', 'net' ]
+CHARTS = {
+    # id: {
+    #     'options': [name, title, units, family, context, charttype],
+    #     'lines': [
+    #         [unique_dimension_name, name, algorithm, multiplier, divisor]
+    #     ]}
+    'filesystem': {
+        'options': ['filesystems', 'Filesystems', 'filesystems', 'filesystem', 'monit.filesystems', 'line'],
+        'lines': []
+    },
+    'directory': {
+        'options': ['directories', 'Directories', 'directories', 'filesystem', 'monit.directories', 'line'],
+        'lines': []
+    },
+    'file': {
+        'options': ['files', 'Files', 'files', 'filesystem', 'monit.files', 'line'],
+        'lines': []
+    },
+    'fifo': {
+        'options': ['fifos', 'Pipes (fifo)', 'pipes', 'filesystem', 'monit.fifos', 'line'],
+        'lines': []
+    },
+    'program': {
+        'options': ['programs', 'Programs statuses', 'programs', 'applications', 'monit.programs', 'line'],
+        'lines': []
+    },
+    'process': {
+        'options': ['processes', 'Processes statuses', 'processes', 'applications', 'monit.services', 'line'],
+        'lines': []
+    },
+    'process_uptime': {
+        'options': ['processes uptime', 'Processes uptime', 'seconds', 'applications', 'monit.process_uptime', 'line', 'hidden'],
+        'lines': []
+    },
+    'process_threads': {
+        'options': ['processes threads', 'Processes threads', 'threads', 'applications', 'monit.process_threads', 'line'],
+        'lines': []
+    },
+    'process_children': {
+        'options': ['processes childrens', 'Child processes', 'childrens', 'applications', 'monit.process_childrens', 'line'],
+        'lines': []
+    },
+    'host': {
+        'options': ['hosts', 'Hosts', 'hosts', 'network', 'monit.hosts', 'line'],
+        'lines': []
+    },
+    'host_latency': {
+        'options': ['hosts latency', 'Hosts latency', 'milliseconds/s', 'network', 'monit.hosts', 'line'],
+        'lines': []
+    },
+    'net': {
+        'options': ['interfaces', 'Network interfaces and addresses', 'interfaces', 'network', 'monit.networks', 'line'],
+        'lines': []
+    },
+}
+
+class Service(UrlService):
+    def __init__(self, configuration=None, name=None):
+        UrlService.__init__(self, configuration=configuration, name=name)
+        self.url = self.configuration.get('url', "http://127.0.0.1:2812/_status?format=xml&level=full")
+        self.order = ORDER
+        self.definitions = CHARTS
```

Comment:
copy `ORDER` and deepcopy `CHARTS`

Reply:
I'm sorry, but i don't understand what you mean. I'm new in python, learned 2 days ago just for purpose of writing this plugin. Could you, please, explain little more details?

Reply:
You meant something like this?

```
import copy

self.order = copy.copy(ORDER)
self.definitions = copy.deepcopy(CHARTS)
```

Reply:
I was wrong, actually it is not needed :smile:

Reply:
Ok, so that's all?

Reply:
Well, as i said earlier - `get_data` method is hard to read (don't you think so?), can you simplify it?

Reply:
Hmm ... Let's make sure we "dot the i's and cross the t's". I'm not a developer, in fact i'm sysadm/sysop/etc. So, if you would be so kind and guide me - i'd be glad to make required changes. In other way - i'll be "shooting in the dark" and wasting both your's and mine time.

Could, please, you provide little more details on this matter?

Reply:
> I'm not a developer

So do i.

> Could you, please, provide little more details on this matter?

Ok, use `format` instead of `%`. Then it will be acceptable.

Reply:
Ok, done. Please, check it.